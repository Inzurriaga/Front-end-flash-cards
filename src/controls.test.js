import React from 'react';
import Controls from './Controls.js';
import { shallow } from 'enzyme';

const create = jest.fn();

describe('controls', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Controls create={create}/>);
  });

  it('should display controls', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should invoke the start method when clicked', () => {
    wrapper.find('.start-button').simulate('click');
    expect(create).toBeCalled();
  });
});