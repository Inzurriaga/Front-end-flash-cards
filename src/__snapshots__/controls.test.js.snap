// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controls should display controls 1`] = `
<div>
  <div
    className="controls-component"
  >
    <input
      className="checkbox"
      id="html"
      name="pick-catagory"
      type="checkbox"
      value="html"
    />
    <label
      className="card html-button"
      for="html"
    >
      <h3>
        HTML
      </h3>
      <img
        src="heart.png"
      />
    </label>
    <input
      className="checkbox"
      id="css"
      name="pick-catagory"
      type="checkbox"
      value="css"
    />
    <label
      className="card css-button"
      for="css"
    >
      <h3>
        CSS
      </h3>
      <img
        src="diamond.png"
      />
    </label>
    <input
      className="checkbox"
      id="javascript"
      name="pick-catagory"
      type="checkbox"
      value="javascript"
    />
    <label
      className="card javascript-button"
      for="javascript"
    >
      <h3>
        Javascript
      </h3>
      <img
        src="spade.png"
      />
    </label>
    <input
      className="checkbox"
      id="personal"
      name="pick-catagory"
      type="checkbox"
      value="personal"
    />
    <label
      className="card personal-button"
      for="personal"
    >
      <h3>
        Personal
      </h3>
      <img
        src="clover.png"
      />
    </label>
  </div>
  <button
    className="start-button"
    onClick={[MockFunction]}
  >
    start undefined
  </button>
</div>
`;
